include/swift/AST/ProtocolConformance.h

```diff
@@ -266,7 +266,11 @@ class alignas(1 << DeclAlignInBits) ProtocolConformance {
   /// Determine whether the witness for the given requirement
   /// is either the default definition or was otherwise deduced.
   bool usesDefaultDefinition(AssociatedTypeDecl *requirement) const;
-  
+
+  /// Returns true if this conformance should be serialized, based on the
+  /// type/protocol involved in it.
+  bool shouldBeSerialized() const;
```

Comment:
Nitpick: the AST doesn't "know" about serialization. Maybe call this hasFixedLayout to be consistent with types